packages/daemon/src/daemon.js

```diff
@@ -689,55 +704,94 @@ const makeDaemonCore = async (
   };
 
   /**
+   * @param {string} [specifiedFormulaNumber] - The formula number of the pet store.
    * @returns {Promise<{ formulaIdentifier: string, value: import('./types').PetStore }>}
    */
-  const incarnatePetStore = async () => {
-    const formulaNumber = await randomHex512();
+  const incarnatePetStore = async specifiedFormulaNumber => {
+    const formulaNumber =
+      specifiedFormulaNumber ??
+      (await formulaGraphLock.enqueue(() => randomHex512()));
```

Comment:
do we need to enqueue the number generation?

Reply:
tl;dr: Yes. I just updated the description of this PR, which explains why.